core/src/main/java/org/geysermc/geyser/registry/populator/BlockRegistryPopulator.java

```diff
@@ -355,6 +356,24 @@ private static void registerJavaBlocks() {
         BlockRegistries.CLEAN_JAVA_IDENTIFIERS.set(cleanIdentifiers.toArray(new String[0]));
 
         BLOCKS_JSON = blocksJson;
+
+        JsonNode blockInteractionsJson;
+        try (InputStream stream = GeyserImpl.getInstance().getBootstrap().getResource("mappings/interactions.json")) {
+            blockInteractionsJson = GeyserImpl.JSON_MAPPER.readTree(stream);
+        } catch (Exception e) {
+            throw new AssertionError("Unable to load Java block interaction mappings", e);
+        }
+
+        BlockRegistries.INTERACTIVE.set(toBlockStateSet((ArrayNode) blockInteractionsJson.get("always_consumes")));
+        BlockRegistries.INTERACTIVE_MAY_BUILD.set(toBlockStateSet((ArrayNode) blockInteractionsJson.get("requires_may_build")));
+    }
+
+    private static IntSet toBlockStateSet(ArrayNode node) {
+        IntSet blockStateSet = new IntOpenHashSet(node.size());
+        for (JsonNode javaIdentifier : node) {
+            blockStateSet.add(BlockRegistries.JAVA_IDENTIFIERS.get().getInt(javaIdentifier.textValue()));
```

Comment:
It might be more convenient to save the mappings here to their block state number?

Reply:
As in saving the block state number in the JSON? I think it is better to only have the identifier in the JSON as it would probably be easier to diff when new blocks are added.
```
	"always_consumes": {
		"minecraft:dispenser[facing=north,triggered=true]": 464,
		"minecraft:dispenser[facing=north,triggered=false]": 465,
		"minecraft:dispenser[facing=east,triggered=true]": 466,
		"minecraft:dispenser[facing=east,triggered=false]": 467,
		"minecraft:dispenser[facing=south,triggered=true]": 468,
		"minecraft:dispenser[facing=south,triggered=false]": 469,
		"minecraft:dispenser[facing=west,triggered=true]": 470,
		"minecraft:dispenser[facing=west,triggered=false]": 471,
```